Framework/Core/include/Framework/MessageContext.h

```diff
@@ -145,6 +145,65 @@ class MessageContext
     }
   };
 
+  // A memory resource which can force a minimum alignment, so that
+  // the whole polymorphic allocator business is happy...
+  class AlignedMemoryResource : public pmr::FairMQMemoryResource
+  {
+   public:
+    AlignedMemoryResource(fair::mq::FairMQMemoryResource* other)
+      : mOther(other)
+    {
+    }
+
+    AlignedMemoryResource(AlignedMemoryResource const& other)
+      : mOther(other.mOther)
+    {
+    }
+
+    bool isValid()
+    {
+      return mOther != nullptr;
+    }
+    FairMQMessagePtr getMessage(void* p) override
+    {
+      return mOther->getMessage(p);
+    }
+
+    void* setMessage(FairMQMessagePtr fmm) override
+    {
+      return mOther->setMessage(std::move(fmm));
+    }
+
+    FairMQTransportFactory* getTransportFactory() noexcept override
+    {
+      return mOther->getTransportFactory();
+    }
+
+    size_t getNumberOfMessages() const noexcept override
+    {
+      return mOther->getNumberOfMessages();
+    }
+
+   protected:
+    void* do_allocate(size_t bytes, size_t alignment) override
+    {
+      return mOther->allocate(bytes, alignment < 64 ? 64 : alignment);
+    }
+
+    void do_deallocate(void* p, size_t bytes, size_t alignment) override
+    {
+      return mOther->deallocate(p, bytes, alignment < 64 ? 64 : alignment);
+    }
+
+    bool do_is_equal(const pmr::memory_resource& other) const noexcept override
+    {
+      return this == &other;
+    }
+
+   private:
+    fair::mq::FairMQMemoryResource* mOther = nullptr; // do you think they'll try to break my balls?
```

Comment:
// do you think they'll try to break my balls?
probably :), I'm not getting this - why encapsulate here? The MemoryResource API is already ok, so FairMQMemoryResource is suitable so why derive AND encapsulate? I have not see the previous iterations so could be some subtle reason :)

Reply:
in the good practice of answering one of my questions: we could indeed use a derived resource to change the default minAlignment so as not to have to specify it in each call, but still not seeing why the encapsulation is there...

Reply:
> // do you think they'll try to break my balls?
> probably :),

For the record, the original was:

```
FairMQMemoryResource *mOther; // do you think they'll try to break my balls?
```

which made the Pink Floyd quote much more sophisticated. The `nullptr` spoiled it..

Reply:
The problem is that the resource is created by the transport, as a `ChannelResource`, passed as allocator to the vector / container (where I cannot control the alignment) and then passed back to FairMQ when sending (with associated `dynamic_cast` to FairMQResource inside FairMQ). I tried to inherit from ChannelResource, override and use the move constructor from the old resource to the new one, but somehow that does not work, I suspect because FairMQ holds the memory resource so moving it creates issues.

Reply:
So basically:
* Encapsulating is needed to override the alignment
* Deriving is needed to send the message

Reply:
ah ok, thanks. Then in terms of the "standard" mOther is the upstream resource (or fallback). All perfectly as it should be:)

Reply:
yeah, but mUpstream would have not allowed for any pink floyd quote. ;-) Kidding apart, I can rename...